### DISCUSSION TITLE
AMA - Community Powered by Discussions

### DISCUSSION THREAD

**Discussion by tuves** [community/community#27346] [Announcements]
## We're hosting an AMA here in the Community!
![AMA Banner](https://user-images.githubusercontent.com/49915749/181312206-b257d9f5-8f19-440b-b48d-6cc59a65d2c8.png)

We've just launched this brand new Community space--and as a housewarming, we want to know you, answer any burning questions you might have, and close out launch week with an async, team AMA (Ask Me Anything)

- This discussion will be open Thursday, July 28th at 9 am PST and close Friday, July 29th at 9 am PST
- During that window, drop your questions in, and our team will rotate in our working hours to answer them as best we can
- Remember that this will all be asynchronous, and we're a mighty team of fewer than 10 people, so please be patient with us 😃 

**Housekeeping**
- If you have specific [feedback/feature requests](https://github.com/community/community), please share them in the appropriate product categories
- If you have questions re: our product roadmap, check out our [public roadmap](https://github.com/github/roadmap)
- Please keep questions/comments clean and friendly, refer to our [Code of Conduct](https://github.com/community/community/blob/e76c2ecc5883c65c889caef6d9e63c99c09eeb53/CODE_OF_CONDUCT.md) + [Community Guidelines](https://docs.github.com/articles/github-community-guidelines) if you have any concerns

**Comment by tuves** [comment]
Hello hello GitHub Community! Welcome to our first Community AMA in our new 🏡!

**Reply by queenofcorgis1** [reply]
Let's get this party started 🎉 :octocat:

**Comment by ChristopherHX** [comment]
Why are all unanswered discussions from the old github.community gone after migration?
I ask because the old discourse has been taken offline and redirects here, but I can't find all of my old posts.
_See this query https://github.com/community/community/discussions/categories/actions-and-packages?discussions_q=category%3A%22Actions+and+Packages%22+created%3A%3C%3D2020-01-01+is%3Aunanswered_
If you remove unanswered from the query then you get a list of migrated discussions.

**Reply by ettaboyle** [reply]
Hi @ChristopherHX we made the decision to only migrate over solutions from the Discourse forum, not all posts. This was an intentional choice that included redirects for all the solutions migrated. All content that was not migrated now links to this discussion homepage. If those posts had marked solutions they should be here - you can also check in the [Wayback Machine](https://archive.org/web/).

**Reply by ChristopherHX** [reply]
Thanks for the answer. So some of my provided answers, which aren't marked as solution by the author are lost.

> you can also check in the [Wayback Machine](https://archive.org/web/).

_I checked the wayback machine, but almost no discussions from the Discourse forum were captured or have much less comments than expected. However the homepage of the Discourse forum has been captured just fine._

**Comment by airtower-luna** [comment]
How did you do the migration at the technical level? I assume there was a fair bit of engineering and testing involved. I love hearing stories of weird bugs, strange discoveries while debugging, surprisingly elegant solutions, and the like, so if you have any please share! :smile_cat:

**Reply by Gisgar3** [reply]
I'd love to know this too. Interested in knowing the technicality behind the process!

**Reply by tuves** [reply]
With the help of folks on our Digital Customer Success team and CS Analytics team, we sorted through categories and accepted solutions, and migrated them over to Discussions. @jricciardi was at the helm of it all, so I'll let him share the more technical aspects of it! I will say that it took a village for such a smooth transition!

**Reply by jfullstackdev** [reply]
wow nice migration

**Reply by jricciardi** [reply]
I'm not an engineer, but I'm happy to share a few aspects of this! The nuts-n-bolts of the migration itself will probably be unsurprising: Discourse provides excellent SQLdb backups which we were able to use to parse, organize, filter, and shape into json for import. The technical side of migration was thankfully straightforward, but it was a journey to arrive at the final 'shape'. 

A lot of time was spent designing a reasonable end-user experience for us to work backward from to arrive at an implementation. Here are some of the more interesting considerations we had to work through:

1. How much content should we bring & what gets priority?
2. What should migrated content actually _look_ like? Do we have to treat it differently?
3. What should we do with content created by users who have Discourse accounts, but no longer have GitHub accounts (you might notice some 'ghost' users around some old solutions) — and how do we handle comments where other users interacted with those now-ghost users?
4. How much [existing GitHub migration tooling](https://github.com/github/gh-gei) can we reasonably apply here?
5. How can we make sure the redirect experience for solutions doesn't block the literal _millions_ of developers who visit thousands of existing solutions every month for troubleshooting help?
6. How should we map content to new categories & repos & tags? (and what should those be?)
7. How many/which solutions should surface in other parts of the overall GitHub experience?

I hope that provides some insight 😄

**Comment by airtower-luna** [comment]
Oh, a silly one: I heard there are achievements for answering discussions. Will we get those applied for the imported answers, too? :smile_cat:

**Reply by tuves** [reply]
Not silly at all, @airtower-luna! @jdevstatic is correct, the imported answers will count towards your answered Discussion count. I just noticed you don't have the galaxy brain badge--which I think you should have, given your solution count. Let me look into this a bit further.

**Reply by jfullstackdev** [reply]
actually the solutions count is missing just this time around 😅 maybe they are being updated or for another release

![image](https://user-images.githubusercontent.com/47092464/181778381-db71befe-f6ea-43f2-858e-dfc8798ca476.png)

**Reply by jfullstackdev** [reply]
> you don't have the galaxy brain badge--which I think you should have, given your solution count. Let me look into this a bit further.

as for the Galaxy Brain badge, it's not retro, rather, it's from the release date onwards, so I just have 1, if it can be retro like the Starstruck badge then that will be be better because it will include all the past valid answers

**Comment by jfullstackdev** [comment]
the solutions we had are already reflected here, I now have more than 100 answered questions because of that, but I think the awesome badges should be included too, like our Top 10 Community Cleanup badge, the Level 3 Regular badge, the Advisory Council badge

hoping this will be added, I'm really up to it

**Reply by tuves** [reply]
Thanks for the feedback, @jdevstatic. We want to ensure that all badges displayed are ones that anyone in the community can receive. I'll share your feedback with the team, however, my gut response is that some of the older Discourse badges will not transfer over here/be re-created.

**Reply by jfullstackdev** [reply]
yup, there were too many badges there, but those I think, just the 3 badges will be enough

Top 10 Community Cleanup badge, the Level 3 Regular badge, the Advisory Council badge

thanks

**Comment by novium2** [comment]
Retiring this post as it is over 6 months old. If you have any questions, consider starting a [new post](https://github.com/orgs/community/discussions/new/choose).
